### PR TITLE
feat(linter): implement unicorn/no-negation-in-equality-check

### LINKED ISSUE #3869: feat(linter): unicorn/no-negation-in-equality-check
To get started, run
```sh
just new-unicorn-rule no-negation-in-equality-check
```

- [Rule docs](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v54.0.0/docs/rules/no-negation-in-equality-check.md)
- [Rule source](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v54.0.0/rules/no-negation-in-equality-check.js)

### PR DESCRIPTION
**Pull request by niss36**
Fixes https://github.com/oxc-project/oxc/issues/3869

Hey there, thought I'd give this a try as it's tagged "good first issue" :) Let me know if there's anything that needs to change, I wasn't sure about the rule category so left as TODO.

### CODE REVIEW

File: crates/oxc_linter/src/snapshots/no_negation_in_equality_check.snap [hunk 1]
```diff
@@ -0,0 +1,93 @@
+---
+source: crates/oxc_linter/src/tester.rs
+---
+  ⚠ eslint-plugin-unicorn(no-negation-in-equality-check): Negated expression is not allowed in equality check.
+   ╭─[no_negation_in_equality_check.tsx:1:1]
+ 1 │ !foo === bar
```
Comment (mysteryven) [inline crates/oxc_linter/src/snapshots/no_negation_in_equality_check.snap:6]: Would you like to point out on the negated unary expression, it's where the error occurs.

File: crates/oxc_linter/src/snapshots/no_negation_in_equality_check.snap [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+   · ────────────
+   ╰────
+  help: Remove the negation operator and use '!==' instead.
```
Comment (mysteryven) [inline crates/oxc_linter/src/snapshots/no_negation_in_equality_check.snap:9]: 🤔 What about change to:
```
 help: Remove the negation operator and use '!==' to instead `===`.
 ```

File: crates/oxc_linter/src/rules/unicorn/no_negation_in_equality_check.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    /// if (!foo !== bar) {}
+    ///
+    /// // Good
+    ///
+    /// if (foo !== bar) {}
+    ///
+    /// if (!(foo === bar)) {}
+    /// ```
+    NoNegationInEqualityCheck,
+    nursery, // TODO: change category to `correctness`, `suspicious`, `pedantic`, `perf`, `restriction`, or `style`
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/unicorn/no_negation_in_equality_check.rs:47]: `correctness`? Since it was picked as recommend rules.

**Review by mysteryven — APPROVED** [review]
Nice work!

**Comment by niss36** [comment]
Thank you! 😄

**Pull request merged**